### PR TITLE
Iso sides

### PR DESCRIPTION
**Pull request by jordansread**
WIP for #290 

This is a partial for that, because it 
- [x] ~~doesn't include putting logs in the sides (they are in `view.x.y.$window` still)~~ 
- [x] ~~breaks a number of tests right now (~8 failures)~~
- [x] ~~changes the `logged` functions to return `sides.n` instead of `sides.x.y` which is probably an API-breaker for some if they were using that~~ this is known, but would like someone to weigh in on it
- [ ] doesn't use the `yaxs` and `xaxs` args. The logic for those is still in the PR, but is commented out. I wasn't sure what was going on before
- [x] ~~doesn't include documentation on the new changes/function~~
- [x] ~~no tests for the new stuff~~
- [x] ~~sides aren't used in the rending of axis, so we still have overplotting~~.
- [x] ~~breaks `draw_custom_grid` which is based on windows, not sides~~
- [x] ~~grid is failing when you don't have a side defined for it (e.g., have 1,2 and want axis side 3 w/ grid)~~
- [x] ~~labels are still in window and not sides~~

But, this does get us most of the way to #290 issue, and also I think addresses some speed issues I ran into when creating very large lists. **now**, the `set_side_lim()` (was `set_view_lim()`) only extracts args from the most recent addition to the list, instead of all (which is what was happening before). It also short-circuits the limit extraction when the user limits are fully defined (min and max). I think both of these additions will make it quite a bit faster, but I haven't tested. 

Also, I noticed that the window$log function isn't really working as expected, and I don't think it has anything to do with this PR. For example, when you have view.1.2 and view.3.2 and log y for one of those, it doesn't also log y for the other view. This causes one of the test failures due to the way I am using `logged` now. 

feedback welcome. I will keep this open while I chip away at some of the remaining pieces.

### CODE REVIEW

File: R/calc_views.R [hunk 1]
```diff
@@ -123,73 +105,6 @@ set_view_lab <- function(views){
...
-      
-      data.lim <- range(data.var[is.finite(data.var)])
-      usr.lim <- views[[n.i]][['window']][[lim.name]][1:2]
-      views[[n.i]][['window']][[lim.name]] <- data.lim
-      views[[n.i]][['window']][[lim.name]][!is.na(usr.lim)] <- usr.lim[!is.na(usr.lim)]
-    
-      
-      usr.axs <- axs[[axs.name]][[n.i]]
-      
-      if (any(!is.na(usr.axs)) && any(usr.axs == 'o')) {
```
Comment (jordansread) [inline R/calc_views.R:171]: This is part that I didn't really get. @lindsaycarr can you help me think of how to pull this into the `sides` model?

File: R/access-gsplot.R [hunk 2]
```diff
@@ -12,14 +12,28 @@ xlim <- function(object, side) UseMethod("xlim")
 
 #' @export
 xlim.gsplot <- function(object, side=NULL){
+  side.lim(object, side, 'x')
+}
 
+side.lim <- function(object, side, axis = c('x','y')){
```
Comment (lindsayplatt) [inline R/access-gsplot.R]: move down so that x and y are next to each other

File: R/access-gsplot.R [hunk 3]
```diff
@@ -12,14 +12,28 @@ xlim <- function(object, side) UseMethod("xlim")
...
+  else {
+    sides <- as.side(names(sides(object)))
+    if (axis == 'y')
+      use.sides <- sides %% 2 == 0
+    else 
+      use.sides <- !sides %% 2 == 0
+    side.names <- as.side_name(sides[use.sides])
+  }
+  
+  lims <- lapply(side.names, function(x) object[[x]]$lim) %>%
```
Comment (lindsayplatt) [inline R/access-gsplot.R]: don't use %>% inside of the function

File: R/calc_views.R [hunk 4]
```diff
@@ -23,42 +19,35 @@ calc_views <- function(gsplot){
...
-  views <- views(gsplot) # existing
-  non.views <- non_views(gsplot)
   add_sides <- set_sides(tail.gs[['gs.config']][['side']])
-    
+  non.views <- non_views(gsplot, include.sides = FALSE)
+  vew.n.sde <- gsplot[c(which_views(gsplot), which_sides(gsplot))]
   if (!is.null(add_sides)){
+    vew.n.sde <- append_sides(vew.n.sde, add_sides)
     to_draw <- setNames(list(c(tail.gs[['arguments']], legend.name=tail.gs[['gs.config']][['legend.name']])), tail.nm)
+    view.name <- sprintf('view.%s.%s',add_sides[1],add_sides[2])
```
Comment (lindsayplatt) [inline R/calc_views.R]: use `as.view_name` fucntion

File: R/callouts.R [hunk 5]
```diff
@@ -45,8 +45,7 @@ callouts.gsplot <- function(object, ..., side=c(1,2)){
 callouts.default <- function(x, y=NULL, labels=NA, length=0.1, angle='auto', cex=par()$cex, ...){
   
   if (is.null(y)) {
-    warning("y=NULL not currently supported in callouts.default")
-    return()
+    stop("y=NULL not currently supported in callouts.default", call. = FALSE)
```
Comment (lindsayplatt) [inline R/callouts.R]: revert please

**Pull request merged**